src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -2060,15 +2060,20 @@ abstract class RefChecks extends Transform {
             stats.foreach { t => if (!checkInterestingResultInStatement(t)) checkPure(t, supple = false) }
             if (result0.nonEmpty) checkPure(result0, supple = true)
 
-            def checkImplicitlyAdaptedBlockResult(t: Tree): Unit =
-              expr match {
-                case treeInfo.Applied(f, _, _) if f.symbol != null && f.symbol.isImplicit =>
-                  f.symbol.paramLists match {
-                    case (p :: Nil) :: _ if p.isByNameParam => refchecksWarning(t.pos, s"Block result was adapted via implicit conversion (${f.symbol}) taking a by-name parameter", WarningCategory.LintBynameImplicit)
-                    case _ =>
-                  }
-                case _ =>
-              }
+            def checkImplicitlyAdaptedBlockResult(t: Tree): Unit = {
+              def loop(t: Tree): Unit =
+                t match {
+                  case Apply(coercion, _) if t.isInstanceOf[ApplyImplicitView] =>
+                    coercion.symbol.paramLists match {
+                      case (p :: Nil) :: _ if p.isByNameParam => refchecksWarning(t.pos, s"Block result was adapted via implicit conversion (${coercion.symbol}) taking a by-name parameter", WarningCategory.LintBynameImplicit)
```

Comment:
```suggestion
                      case (p :: Nil) :: _ if p.isByNameParam => refchecksWarning(t.pos, s"Block result was adapted via implicit conversion (${coercion.symbol}) taking a by-name parameter; statements are not evaluated eagerly", WarningCategory.LintBynameImplicit)
```

or something like that?

Reply:
Or just the opposite. This is the opposite where only the result expression is adapted. I'll upgrade both messages!